File: WEEK1/DAY5/Prog5/P5.java
import java.util.*;
public class P5 {
    public static boolean prime(int n){
        if(n<2){
            return false;
        }
        if(n==2 || n==3){
            return true;
        }
        if(n%2==0 || n%3==0){
            return false;
        }
        int i=5;
        while(i<=(int)Math.pow(n, 0.5)){
            if(n%i==0 || n%(i+2)==0){
                return false;
            }
            i+=6;
        }
        return true;
    }
    public static void main(String[] args){
        int N;
        Scanner sc=new Scanner(System.in);
        System.out.println("Count primes between 1 and N\n");
        System.out.print("Enter N: ");
        N=sc.nextInt();
        int cnt=0;
        for(int i=2;i<=N;i++){
            if(prime(i)){
                cnt+=1;
            }
        }
        System.out.println("There are "+cnt+" prime numbers");
        sc.close();
    }
}

File: WEEK1/DAY5/Prog5/P5.c
#include <stdio.h>
#include <math.h>
int prime(int n){
    if(n<2){
        return 0;
    }
    if(n==2 || n==3){
        return 1;
    }
    if(n%2==0 || n%3==0){
        return 0;
    }
    int i=5;
    while(i<(int)pow(n,0.5)){
        if(n%i==0 || n%(i+2)==0){
            return 0;
        }
        i+=6;
    }
    return 1;
}
int main(){
    int N;
    printf("Count primes between 1 and N\n");
    printf("Enter N: ");
    scanf("%d",&N);
    int cnt=0;
    for(int i=2;i<=N;i++){
        if(prime(i)==1){
            cnt+=1;
        }
    }
    printf("There are %d prime numbers\n",cnt);
}